index.js

```diff
@@ -19,170 +13,165 @@ const npar = spec => {
   }
   return spec
 }
+const mapJSON = (value, [key]) => {
+  if (value === 'read') {
+    return [key, 'read-only']
+  } else if (value === 'write') {
+    return [key, 'read-write']
+  } else {
+    return [key, value]
+  }
+}
 
 const cmd = module.exports = {}
 
 cmd.public = (spec, opts) => setAccess(spec, 'public', opts)
 cmd.restricted = (spec, opts) => setAccess(spec, 'restricted', opts)
-function setAccess (spec, access, opts) {
-  opts = AccessConfig(opts)
-  return pwrap(opts, () => {
+function setAccess (spec, access, opts = {}) {
+  return Promise.resolve().then(() => {
     spec = npar(spec)
     validate('OSO', [spec, access, opts])
     const uri = `/-/package/${eu(spec.name)}/access`
-    return npmFetch(uri, opts.concat({
+    return npmFetch(uri, {
+      ...opts,
       method: 'POST',
       body: { access },
       spec
-    }))
-  }).then(res => res.body.resume() && true)
```

Comment:
More of an idle question than feedback per se, but should we be concerned that `res.body` is not getting flushed any more?  I wonder what that's doing there?